yapf/yapflib/unwrapped_line.py

```diff
@@ -227,6 +227,76 @@ def _IsUnaryOperator(tok):
   return format_token.Subtype.UNARY_OPERATOR in tok.subtypes
 
 
+def _IsMOperator(leaf):
+  """ See definition of an m_expr in the python reference:
+  https://docs.python.org/3/reference/expressions.html#binary-arithmetic-operations
+  """
+  return leaf.value in ["*", "@", "//", "%", "/"]
+
+
+def _IsAOperator(leaf):
+  """ See definition of an a_expr in the python reference:
+  https://docs.python.org/3/reference/expressions.html#binary-arithmetic-operations
+  """
+  return leaf.value in ["+", "-"]
+
+
+def _IsBinaryArithmeticOperator(leaf):
+  return _IsMOperator(leaf) or _IsAOperator(leaf)
+
+
+def _IsIndependentOperator(leaf):
+  """Tests whether the operator's operands are leaf nodes."""
+  siblings = leaf.parent.children
+  return all(map(lambda s: len(s.children) == 0, siblings))
```

Comment:
I'm not a fan of operating on `pytree` nodes at this level. I would eventually like to replace lib2to3's parser with something better. I think it would be nicer to add a `SIMPLE_EXPRESSION` (or something more appropriately named) subtype and assign in `subtype_assigner.py`.

Reply:
Created subtype `SIMPLE_EXPRESSION`.